File: backend/app/models/enums.py
from enum import Enum


class Step(str, Enum):
    MIGRATION_FAILED = "migration_failed"  # Interactive - blocks app until resolved
    WELCOME = "welcome"  # Interactive - shown on very first run, before ABS setup
    ABS_SETUP = "abs_setup"  # Interactive
    LLM_SETUP = "llm_setup"  # Interactive
    ASR_SETUP = "asr_setup"  # Interactive
    IDLE = "idle"  # Interactive
    VALIDATING = "validating"
    DOWNLOADING = "downloading"
    FILE_PREP = "file_prep"
    SELECT_WORKFLOW = "select_workflow"  # Interactive
    AUDIO_ANALYSIS = "audio_analysis"
    VAD_PREP = "vad_prep"
    VAD_ANALYSIS = "vad_analysis"
    INTELLIGENT_CHAPTER_DETECTION = "intelligent_chapter_detection"  # Interactive
    VOSK_ANALYSIS = "vosk_analysis"
    LLM_CANDIDATE_TRIAGE = "llm_candidate_triage"
    PARTIAL_SCAN_PREP = "partial_scan_prep"
    PARTIAL_AUDIO_ANALYSIS = "partial_audio_analysis"
    PARTIAL_VAD_ANALYSIS = "partial_vad_analysis"
    INITIAL_CHAPTER_SELECTION = "initial_chapter_selection"  # Interactive
    CONFIGURE_ASR = "configure_asr"  # Interactive
    AUDIO_EXTRACTION = "audio_extraction"
    TRIMMING = "trimming"
    ASR_PROCESSING = "asr_processing"
    CHAPTER_EDITING = "chapter_editing"  # Interactive
    AI_CLEANUP = "ai_cleanup"
    REVIEWING = "reviewing"  # Interactive
    COMPLETED = "completed"  # Interactive

    @property
    def ordinal(self):
        return list(self.__class__.__members__).index(self.name)


class DetectionMode(str, Enum):
    """Detection strategy for Smart Detect / Realign workflows"""

    STANDARD = "standard"
    AUTO = "auto"
    DRAMATIZED = "dramatized"


class RestartStep(str, Enum):
    """Potential restart points for session restart"""

    IDLE = Step.IDLE.value
    SELECT_WORKFLOW = Step.SELECT_WORKFLOW.value
    INTELLIGENT_CHAPTER_DETECTION = Step.INTELLIGENT_CHAPTER_DETECTION.value
    INITIAL_CHAPTER_SELECTION = Step.INITIAL_CHAPTER_SELECTION.value
    CONFIGURE_ASR = Step.CONFIGURE_ASR.value
    CHAPTER_EDITING = Step.CHAPTER_EDITING.value

    @property
    def ordinal(self):
        return list(self.__class__.__members__).index(self.name)

File: frontend/src/types/enums.ts
export const Step = {
  MIGRATION_FAILED: 'migration_failed',
  WELCOME: 'welcome',
  ABS_SETUP: 'abs_setup',
  LLM_SETUP: 'llm_setup',
  ASR_SETUP: 'asr_setup',
  IDLE: 'idle',
  VALIDATING: 'validating',
  DOWNLOADING: 'downloading',
  FILE_PREP: 'file_prep',
  SELECT_WORKFLOW: 'select_workflow',
  AUDIO_ANALYSIS: 'audio_analysis',
  VAD_PREP: 'vad_prep',
  VAD_ANALYSIS: 'vad_analysis',
  INTELLIGENT_CHAPTER_DETECTION: 'intelligent_chapter_detection',
  VOSK_ANALYSIS: 'vosk_analysis',
  LLM_CANDIDATE_TRIAGE: 'llm_candidate_triage',
  PARTIAL_SCAN_PREP: 'partial_scan_prep',
  PARTIAL_AUDIO_ANALYSIS: 'partial_audio_analysis',
  PARTIAL_VAD_ANALYSIS: 'partial_vad_analysis',
  INITIAL_CHAPTER_SELECTION: 'initial_chapter_selection',
  CONFIGURE_ASR: 'configure_asr',
  AUDIO_EXTRACTION: 'audio_extraction',
  TRIMMING: 'trimming',
  ASR_PROCESSING: 'asr_processing',
  CHAPTER_EDITING: 'chapter_editing',
  AI_CLEANUP: 'ai_cleanup',
  REVIEWING: 'reviewing',
  COMPLETED: 'completed',
} as const;

export type Step = (typeof Step)[keyof typeof Step];

export const RestartStep = {
  IDLE: Step.IDLE,
  SELECT_WORKFLOW: Step.SELECT_WORKFLOW,
  INTELLIGENT_CHAPTER_DETECTION: Step.INTELLIGENT_CHAPTER_DETECTION,
  INITIAL_CHAPTER_SELECTION: Step.INITIAL_CHAPTER_SELECTION,
  CONFIGURE_ASR: Step.CONFIGURE_ASR,
  CHAPTER_EDITING: Step.CHAPTER_EDITING,
} as const;

export type RestartStep = (typeof RestartStep)[keyof typeof RestartStep];

// Detection strategy for the Smart Detect / Realign workflows
export const DetectionMode = {
  STANDARD: 'standard',
  AUTO: 'auto',
  DRAMATIZED: 'dramatized',
} as const;

export type DetectionMode = (typeof DetectionMode)[keyof typeof DetectionMode];
